orte/mca/odls/default/odls_default_component.c

```diff
@@ -68,13 +70,28 @@ orte_odls_base_component_t mca_odls_default_component = {
     },
 };
 
+int orte_odls_default_maxfd = 1024;
 
 
 int orte_odls_default_component_open(void)
 {
     return ORTE_SUCCESS;
 }
 
+int orte_odls_default_component_register(void)
+{
+    mca_base_component_var_register(&mca_odls_default_component.version, "maxfd",
```

Comment:
I don't understand why you made this non-Linux only?  Seems overly complicated.

Reply:
In the Linux case, we call https://github.com/open-mpi/ompi/blob/82b1eb46d4392504a2054960d713bfdfa057da04/orte/mca/odls/default/odls_default_module.c#L298-L334

That routine looks at `/proc/self/fd` to know precisely which fd's to close.  If that routine fails (which it does in the non-Linux case), we fall back to the "for fd = 3..fdmax; close(fd)" approach.  Hence, this fdmax is really only relevant for non-Linux environments.